File: 1855.下标对中的最大距离.go
/*
 * @lc app=leetcode.cn id=1855 lang=golang
 *
 * [1855] 下标对中的最大距离
 */

// @lc code=start
func maxDistance(nums1 []int, nums2 []int) (res int) {
	m, n := len(nums1), len(nums2)
	index := 0
	for i := 0; i < n; i++ {
		for index < m && nums1[index] > nums2[i] {
			index++
		}
		if index < m {
			res = max(res, i-index)
		}
	}
	return
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

// @lc code=end

File: 1855.下标对中的最大距离.py
#
# @lc app=leetcode.cn id=1855 lang=python3
#
# [1855] 下标对中的最大距离
#

# @lc code=start
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        n1, n2 = len(nums1), len(nums2)
        i = 0
        res = 0
        for j in range(n2):
            while i < n1 and nums1[i] > nums2[j]:
                i += 1
            if i < n1:
                res = max(res, j - i)
        return res
# @lc code=end

